packages/drivers/odsp-driver/src/localOdspDocumentStorageManager.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { ISummaryContext } from "@fluidframework/driver-definitions";
+import { UsageError } from "@fluidframework/driver-utils";
+import * as api from "@fluidframework/protocol-definitions";
+import { OdspDocumentStorageServiceBase } from "./odspDocumentStorageManager";
+import { ISnapshotContents } from "./odspPublicUtils";
+import { IOdspSnapshot } from "./contracts";
+import { convertOdspSnapshotToSnapshotTreeAndBlobs } from "./odspSnapshotParser";
+import { parseCompactSnapshotResponse } from "./compactSnapshotParser";
+import { ReadBuffer } from "./ReadBufferUtils";
+
+/**
+ * ODSP document storage service that works on a provided snapshot for all its processing.
+ * Attempting to use unsupported actions/methods will result in errors being thrown.
+ */
+export class LocalOdspDocumentStorageService extends OdspDocumentStorageServiceBase {
+    private snapshotTreeId: string | undefined;
+
+    constructor(
+        private readonly logger: ITelemetryLogger,
+        private readonly localSnapshot: Uint8Array | string,
+    ) {
+        super();
+    }
+
+    private calledGetVersions = false;
+
+    public async getVersions(blobid: string | null, count: number, _scenarioName?: string): Promise<api.IVersion[]> {
+        assert(blobid === null, "Invalid usage. \"blobid\" should always be null");
+        assert(count === 1, "Invalid usage. \"count\" should always be 1");
```

Comment:
We should remove this assert. User can supply any value and the service returns the available ones less than or equal to count. So, if user asked for 10 but server has only 1, then it returns only 1.